### PR TITLE
Error Handling on Read/Write

### PR DESCRIPTION
**Pull request by sjkelly**
Needed to add to #78

### CODE REVIEW

File: src/highlevel.jl [hunk 1]
```diff
@@ -734,6 +734,14 @@ function Base.read!(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout
         # collect list of pointers to pass to SoapySDR
         buff_ptrs = Ref(map(b -> pointer(b, total_nread+1), buffers))
         nread, flags, timens = SoapySDRDevice_readStream(s.d, s, buff_ptrs, samples_to_read - total_nread, timeout_us)
+
+        if nread == SOAPY_SDR_OVERFLOW
+            total_nread += s.mtu
+            continue
+        elseif nread < 0
```
Comment (sjkelly) [inline src/highlevel.jl:741]: Right, this should handle Timeout also..

File: src/highlevel.jl [hunk 2]
```diff
@@ -757,11 +757,11 @@ function flags_to_set(flags)
 end
 
 """
-    read!(s::SoapySDR.Stream{T}, buffers::NTuple{N, Vector{T}}; [timeout])
+    read!(s::SoapySDR.Stream{T}, buffers::NTuple{N, Vector{T}}; [timeout], [flags::Ref{Int}], [throw_error=false])
 
 Read data from the device into the given buffers.
 """
-function Base.read!(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout=nothing) where {N, T}
+function Base.read!(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout=nothing, flags=nothing, throw_error=false) where {N, T}
```
Comment (staticfloat) [inline src/highlevel.jl:764]:
```suggestion
function Base.read!(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout=nothing, flags::Union{Ref{Int},Nothing}=nothing, throw_error::Bool=false) where {N, T}
```

File: src/highlevel.jl [hunk 3]
```diff
@@ -842,11 +851,11 @@ function deactivate!(s::Stream; flags = 0, timens = nothing)
 end
 
 """
-    write(s::SoapySDR.Stream{T}, buffer::NTuple{N, Vector{T}}; [timeout]) where {N, T}
+    write(s::SoapySDR.Stream{T}, buffer::NTuple{N, Vector{T}}; [timeout], [flags::Ref{Int}], [throw_error=false]) where {N, T}
 
 Write data from the given buffers into the device.  The buffers must all be the same length.
 """
-function Base.write(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout = nothing) where {N, T}
+function Base.write(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout = nothing, flags=nothing, throw_error=false) where {N, T}
```
Comment (staticfloat) [inline src/highlevel.jl:858]:
```suggestion
function Base.write(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout = nothing, flags::Union{Ref{Int},Nothing}=nothing, throw_error::Bool=false) where {N, T}
```

**Review by staticfloat — APPROVED** [review]

File: src/highlevel.jl [hunk 4]
```diff
@@ -864,6 +873,15 @@ function Base.write(s::Stream{T}, buffers::NTuple{N, AbstractVector{T}}; timeout
     GC.@preserve buffers while total_nwritten < samples_to_write
         buff_ptrs = Ref(map(b -> pointer(b, total_nwritten+1), buffers))
         nwritten, flags = SoapySDRDevice_writeStream(s.d, s, buff_ptrs, samples_to_write - total_nwritten, 0, 0, timeout_us)
```
Comment (staticfloat) [inline src/highlevel.jl:875]:
```suggestion
        nwritten, out_flags = SoapySDRDevice_writeStream(s.d, s, buff_ptrs, samples_to_write - total_nwritten, 0, 0, timeout_us)
```

**Comment by sjkelly** [comment]
This is still causing the compiler to segfault...

**Comment by staticfloat** [comment]
@sjkelly Can you see if the latest commit fixes this branch for you?

**Comment by sjkelly** [comment]
Ah yes, it works!!!

**Pull request merged**